openedx/core/djangoapps/schedules/signals.py

```diff
@@ -179,3 +170,39 @@ def _should_randomly_suppress_schedule_creation(
         return True
 
     return False
+
+
+def _should_create_schedule(enrollment_created, current_site, enrollment):
+    """Checks configuration and returns whether a Schedule should be created for this enrollment"""
+    if not enrollment_created:
+        # only create schedules when enrollment records are created
+        return False
+
+    if current_site is None:
+        log.debug('Schedules: No current site')
+        return False
+
+    schedule_config = ScheduleConfig.current(current_site)
+    if (
+        not schedule_config.create_schedules and
+        not CREATE_SCHEDULE_WAFFLE_FLAG.is_enabled(enrollment.course_id)
+    ):
+        log.debug('Schedules: Creation not enabled for this course or for this site')
+        return False
+
+    if not enrollment.course_overview.self_paced:
+        log.debug('Schedules: Creation only enabled for self-paced courses')
+        return False
+
+    return True
+
+
+def _get_experience_type(enrollment):
+    """Returns the ScheduleExperience type for the given enrollment
+
```

Comment:
Nit: Can we start the docstring on a new line separate from the beginning quotes?  Also, is there a need for an extra line between these sentences?

Reply:
I've always followed [PEP 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) which says you need a blank line between the summary line and the rest of the docstring. It also says you can have the summary line on the same line as the quotes or the next line, so I'll move this one down.

Reply:
Perhaps you haven't seen the edX [python coding guidelines](http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings), which says we intentionally stray away from PEP-257 regarding the docstring format.

Reply:
You each get one point: The coding guidelines say the docstring should start on its own line (not with the triple-quotes), and that the summary should be separated from the rest of the docstring.  Extra nit: use a period at the end of the summary.

Reply:
👍  yes - @thallada's right about the one-line summary in the beginning.  I guess I hadn't read it too closely as I had thought.